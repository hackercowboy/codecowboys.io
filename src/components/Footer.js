import React from 'react';
import PropTypes from 'prop-types';
import { Parallax } from 'react-parallax';
import { injectIntl } from 'react-intl';

import Section from './Section';

import './Footer.scss';

const background = require('../assets/images/bg-7.jpg');

function Footer({ intl }) {
  return (
    <footer>
      <Parallax
        bgImage={background}
        strength={500}
      >
        <Section dark>
          <div className="footer-quote">
            <i className="fas fa-angle-double-left" aria-hidden="true" />
              &nbsp;What one programmer can do in one month, two programmers can do in two months.&nbsp;
            <i className="fas fa-angle-double-right" aria-hidden="true" />
            <br />
            - Fred Brooks
          </div>
          <div className="footer">
            <div className="footer-copyright">
              ©
              {new Date().getFullYear()}
              {' '}
              David Übelacker
            </div>
            <div className="footer-item">
            &nbsp;
              <a id="imprint-link" href={intl.formatMessage({ id: 'footer.imprint_link' })}>{intl.formatMessage({ id: 'footer.imprint' })}</a>
              &nbsp;
              |
              &nbsp;
              <a id="privacy-link" href={intl.formatMessage({ id: 'footer.privacy_link' })}>{intl.formatMessage({ id: 'footer.privacy' })}</a>
            </div>
            <div className="footer-item">
              <a id="language-switch-en" href="/en">English</a>
              {' '}
              |
              <a id="language-switch-de" href="/de">Deutsch</a>
            </div>
          </div>
        </Section>
      </Parallax>
    </footer>
  );
}

Footer.propTypes = {
  intl: PropTypes.shape({ formatMessage: PropTypes.func.isRequired }).isRequired,
};

export default injectIntl(Footer);
