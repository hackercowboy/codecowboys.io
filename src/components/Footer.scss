.footer {
  min-height: 50vh;
  padding: 20px 10px;
  text-align: center;
  width: 100%;
}

.footer-quote {
  font-size: 40px;
  font-weight: 100;
  padding: 50px 10px;
  text-align: center;
}

.footer-copyright {
  padding: 15px 0;
}

.footer-item {
  padding: 3px 0;
}
