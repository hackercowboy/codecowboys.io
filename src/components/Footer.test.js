import React from 'react';
import { shallow } from 'enzyme';

import Footer from './Footer';

describe('<Footer/>', () => {
  it('should render on server', () => {
    jest.spyOn(Date.prototype, 'getFullYear').mockReturnValue(2050);
    process.browser = false;
    const wrapper = shallow(<Footer />);
    expect(wrapper.shallow().debug()).toMatchSnapshot();
  });

  it('should render on browser', () => {
    jest.spyOn(Date.prototype, 'getFullYear').mockReturnValue(2050);
    process.browser = true;
    const wrapper = shallow(<Footer />);
    expect(wrapper.shallow().debug()).toMatchSnapshot();
  });
});
