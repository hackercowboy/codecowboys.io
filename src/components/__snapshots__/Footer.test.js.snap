// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should render on browser 1`] = `
"<LazyLoad offset={300} className=\\"\\" classNamePrefix=\\"lazyload\\" once={false} overflow={false} resize={false} scroll={true} unmountIfInvisible={false}>
  <footer>
    <Parallax bgImage=\\"/images/bg-7.webp\\" bgImageAlt=\\"Background image shoing the Wave in the Coyote Buttes North area of the Utah Arizona border\\" strength={500} bgClassName=\\"react-parallax-bgimage\\" className=\\"\\" contentClassName=\\"\\" disabled={false}>
      <Section dark={true} id={[undefined]} subtitle={[undefined]} title={[undefined]}>
        <div className=\\"footer-quote\\">
          <i className=\\"fas fa-angle-double-left\\" aria-hidden=\\"true\\" />
           What one programmer can do in one month, two programmers can do in two months. 
          <i className=\\"fas fa-angle-double-right\\" aria-hidden=\\"true\\" />
          <br />
          - Fred Brooks
        </div>
        <div className=\\"footer\\">
          <div className=\\"footer-item\\">
             
            <a id=\\"imprint-link\\" href=\\"/en/imprint\\">
              Imprint
            </a>
              |  
            <a id=\\"privacy-link\\" href=\\"/en/privacy\\">
              Privacy Policy
            </a>
              |  
            <a id=\\"language-switch-en\\" href=\\"/en\\">
              English
            </a>
              |  
            <a id=\\"language-switch-de\\" href=\\"/de\\">
              Deutsch
            </a>
          </div>
          <div className=\\"footer-item social-links\\">
            <a id=\\"github-link\\" href=\\"https://github.com/hackercowboy\\">
              <FaGithub size=\\"2.5em\\" />
            </a>
            <a id=\\"linkedin-link\\" href=\\"https://www.linkedin.com/in/daviduebelacker/\\">
              <FaLinkedin size=\\"2.5em\\" />
            </a>
          </div>
          <div className=\\"footer-copyright\\">
            ©
            2050
             
            David Übelacker
          </div>
        </div>
      </Section>
    </Parallax>
  </footer>
</LazyLoad>"
`;

exports[`<Footer/> should render on server 1`] = `
"<footer>
  <Section dark={true} id={[undefined]} subtitle={[undefined]} title={[undefined]}>
    <div className=\\"footer-quote\\">
      <i className=\\"fas fa-angle-double-left\\" aria-hidden=\\"true\\" />
       What one programmer can do in one month, two programmers can do in two months. 
      <i className=\\"fas fa-angle-double-right\\" aria-hidden=\\"true\\" />
      <br />
      - Fred Brooks
    </div>
    <div className=\\"footer\\">
      <div className=\\"footer-item\\">
         
        <a id=\\"imprint-link\\" href=\\"/en/imprint\\">
          Imprint
        </a>
          |  
        <a id=\\"privacy-link\\" href=\\"/en/privacy\\">
          Privacy Policy
        </a>
          |  
        <a id=\\"language-switch-en\\" href=\\"/en\\">
          English
        </a>
          |  
        <a id=\\"language-switch-de\\" href=\\"/de\\">
          Deutsch
        </a>
      </div>
      <div className=\\"footer-item social-links\\">
        <a id=\\"github-link\\" href=\\"https://github.com/hackercowboy\\">
          <FaGithub size=\\"2.5em\\" />
        </a>
        <a id=\\"linkedin-link\\" href=\\"https://www.linkedin.com/in/daviduebelacker/\\">
          <FaLinkedin size=\\"2.5em\\" />
        </a>
      </div>
      <div className=\\"footer-copyright\\">
        ©
        2050
         
        David Übelacker
      </div>
    </div>
  </Section>
</footer>"
`;
